### PR TITLE
Ron azar/backward gady

### PR DESCRIPTION
**Pull request by RonAzar**

### CODE REVIEW

**Review by brainpicker — COMMENTED** [review]
Can you show some implementation here? How does your API which received Owner until now works now with both versions?
What happens if I send empty name, first name and last name?

File: src/main/kotlin/com/hibob/academy/rest_api/Pet.kt [hunk 1]
```diff
@@ -2,4 +2,4 @@ package com.hibob.academy.rest_api
 
 import java.util.Date
 
-data class Pet(val petId: Long, val name: String, val type: String, val companyId: Int, val dateOfArrival: Date)
\ No newline at end of file
+data class Pet(val petId: Long, val name: String?, val firstName: String?, val lastName: String?, val type: String, val companyId: Int, val dateOfArrival: Date)
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/rest_api/Pet.kt:5]: It was before your exercise...

File: src/main/kotlin/com/hibob/academy/rest_api/PetController.kt [hunk 2]
```diff
@@ -7,16 +7,29 @@ import org.springframework.stereotype.Controller
...
+        val hasName = !pet.name.isNullOrBlank()
+        val hasFirstNameAndLastName = !pet.firstName.isNullOrBlank() && !pet.lastName.isNullOrBlank()
+
+        if (hasName && (pet.firstName != null || pet.lastName != null)) {
+            throw BadRequestException("Pet cannot have both a name and first name/last name.")
+        }
+
+        if (!hasName && !hasFirstNameAndLastName) {
+            throw BadRequestException("Pet must have either a full name or both first name and last name.")
+        }
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/rest_api/PetController.kt]: Got it!

File: src/main/kotlin/com/hibob/academy/rest_api/PetController.kt [hunk 3]
```diff
@@ -71,9 +85,10 @@ class PetResource {
     @PUT
     @Path("/{petId}")
     fun updatePet(@PathParam("petId") petId: Long, updatedPet: Pet): Response {
+        validatePet(updatedPet)
         val index = pets.indexOfFirst { it.petId == petId }
         return if (index >= 0) {
-            pets[index] = updatedPet.copy(petId = petId) // Keep the same pet ID
+            pets[index] = updatedPet.copy(petId = petId)
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/rest_api/PetController.kt]: I dont have dao at this exercise

**Pull request closed without merging**